docs/logs/pipelines.md

```diff
@@ -103,26 +103,38 @@ Field Extractor provides a bunch of predefined patterns you may use for your pur
 ![Processor Grok Field Extractor](../images/logs/pipelines/processor-grok.png)
 
 ##### Scripting
-Simple scripting is supported using Script processor. String processing and basic math operations are supported. You can access fields using `get` method. It also supports storing/reading intermediate results using `vars` method.
-The following is supported:
-
-- Math operators: +, -, /, *, %, ^, e.g. `get('size.kb')*2^10`
-- Relational operators: ==, !=, <, <=, >, >=, e.g. `get('size') > 10`
-- Logical operators: and, or, not, e.g. `get('size') > -1 and get('size') < 10`
-- Ternary operator, e.g. `get('speed') < 1000 ? 'OK' : 'SLOW'`
-- Elvis operator `?:`, e.g. `get('severity')?:'INFO'`
-- String functions, e.g. `get('severity').toUpperCase()` or `get('message').split('-')[3]`
-- vars method, e.g. `vars(idx, get('message').indexOf('-'))` and `get('message').substring(vars('idx'), vars('idx') + 3)`
-
-Conditional block and loops are not supported and each line can have only a single statement. The last line should result in a value that will be stored as a field, while other lines should only be vars statements.
+Scripting is supported using Script processor which uses [painless](https://www.elastic.co/guide/en/elasticsearch/painless/current/painless-guide.html) language. Painless is a simple scripting language similar to Java. It can safely be used for stored scripts.
+You can access fields using `doc` method. Since type of value can be anything before using it you must cast value to specific type like: String, Integer, Double etc.
+
+- Math operators: +, -, /, *, %, ^, e.g. `(Integer)doc['size.kb']*2`
+- Relational operators: ==, !=, <, <=, >, >=, e.g. `(Integer)doc['size'] > 10`
+- Logical operators: &&, ||, !, e.g. `(Integer)doc['size'] > -1 && (Integer)doc['size'] < 10`
+- Ternary operator, e.g. `(Integer)doc['speed'] < 1000 ? "OK" : "SLOW"`
+- String functions, e.g. `((String)doc['severity']).toUpperCase()` or `((String)doc['message']).splitOnToken('-')[3]`
+
+Conditional block and loops are supported. The last line should result in a value that will be stored as a field.
+
+```java
+String[] tokens = ((String)doc['message']).splitOnToken(' ');
+if(tokens.length % 2 != 0){
+  String result = "Iterator ";
+  for(int i = 0; i < tokens.length; i++){
+      result += i;
+  }
+  result;
+}
+else {
+  tokens.length * 10;
+}
+```
```

Comment:
It's unclear to me why we suddenly show this piece of code?  Should we say something before it, say what it is, what it does, or some such, so that this page flows better?

Reply:
It is just simple Artificial example which shows how if/else + for may work.

Reply:
@otisg here is how it looks in doc
![Screenshot from 2021-11-24 09-53-47](https://user-images.githubusercontent.com/1091862/143189608-94c57091-9602-400b-87f3-7694c7111de8.png)
![Screenshot from 2021-11-24 09-53-37](https://user-images.githubusercontent.com/1091862/143189612-cc4d330b-60f0-4bc8-a367-fe9a28bf48ec.png)

Reply:
You maybe have in mind some real example which can be useful?